app/src/api/run-for-effect.lisp

```diff
@@ -0,0 +1,21 @@
+;;;; api/effect-change.lisp
+;;;;
+;;;; Author: Erik Davis
+
+(in-package #:qvm-app)
+
+(defgeneric perform-run-for-effect (simulation-method quil num-qubits &key gate-noise measurement-noise)
+  (:method (simulation-method quil num-qubits &key gate-noise measurement-noise)
+    (declare (ignore gate-noise measurement-noise))
+    (api-method-not-implemented-error 'perform-run-for-effect))
+  (:documentation "Executes a quil program for the purpose of updating a shared or persistent wavefunction."))
+
+(defmethod perform-run-for-effect ((simulation-method (eql 'pure-state)) quil num-qubits &key gate-noise measurement-noise)
+  (unless **persistent-wavefunction**
```

Comment:
Why error? Maybe an warning should be issued that `run-for-effect` is being called on a statelessly instead.

Reply:
Good call, I've switched to using `WARN`.